tests/elements-regression/helper.ts

```diff
@@ -16,37 +16,44 @@ export default class ElementRegressionHelper {
 		this.editorPage = new EditorPage( page, testInfo );
 	}
 
-	async doScreenshotComparison( args: { widgetType: string, hoverSelector: {[ key: string ]: string} } ) {
-		let animation: 'disabled' | 'allow' = 'disabled';
-		const widgetCount = await this.editorPage.getWidgetCount();
-		for ( let i = 0; i < widgetCount; i++ ) {
-			const widget = this.editorPage.getWidget().nth( i );
-			await expect.soft( widget ).not.toHaveClass( /elementor-widget-empty/ );
+	async doScreenshot( widgetType: string, isPublished: boolean ) {
+		if ( widgetType.includes( 'hover' ) ) {
+			return;
+		}
+		const locator = isPublished
+			? this.page.locator( EditorSelectors.container )
+			: this.editorPage.getPreviewFrame().locator( EditorSelectors.container );
 
-			if ( args.widgetType.includes( 'hover' ) ) {
-				await widget.locator( args.hoverSelector[ args.widgetType ] ).hover();
-				animation = 'allow';
-			}
-			await expect.soft( widget )
-				.toHaveScreenshot( `${ args.widgetType }_${ i }.png`, { maxDiffPixels: 200, timeout: 10000, animations: animation } );
+		const label = isPublished ? 'published' : 'editor';
+
+		if ( ! isPublished ) {
+			await this.page.evaluate( async () => {
+				const iframe = document.getElementById( 'elementor-preview-iframe' );
+				iframe.style.height = '3000px';
+			} );
 		}
+		await expect.soft( locator )
+			.toHaveScreenshot( `${ widgetType }_${ label }.png`, { maxDiffPixels: 200, timeout: 10000 } );
 	}
 
-	async doScreenshotPublished( args: { widgetType: string, hoverSelector: {[ key: string ]: string} } ) {
-		const widgetCount = await this.page.locator( EditorSelectors.widget ).count();
+	async doHoverScreenshot( args:Omit<ScreenShot, 'device'> ) {
 		if ( args.widgetType.includes( 'hover' ) ) {
+			const widget = args.isPublished
+				? this.page.locator( EditorSelectors.widget )
+				: this.editorPage.getPreviewFrame().locator( EditorSelectors.widget );
+			const label = args.isPublished ? 'published' : 'editor';
+			const widgetCount = await widget.count();
+
 			for ( let i = 0; i < widgetCount; i++ ) {
-				await this.page.locator( `${ EditorSelectors.widget } ${ args.hoverSelector[ args.widgetType ] }` ).nth( i ).hover();
-				await expect.soft( this.page.locator( EditorSelectors.widget ).nth( i ) ).
-					toHaveScreenshot( `${ args.widgetType }_${ i }_published.png`, { maxDiffPixels: 200, timeout: 10000, animations: 'allow' } );
+				await expect( widget.nth( i ) ).not.toHaveClass( /elementor-widget-empty/ );
```

Comment:
Do we want to have a soft assertion here?
```suggestion
				await expect.soft( widget.nth( i ) ).not.toHaveClass( /elementor-widget-empty/ );
```

Reply:
I don't want to continue if widget failed to load, so no soft here